benchmark/framework/aea_test_wrapper.py

```diff
@@ -261,16 +260,22 @@ def set_fake_connection(
 
         :return: None
         """
+        if self._fake_connection:
+            raise Exception("Fake connection is alrady set!")
+
         envelope = envelope or self.dummy_envelope()
-        self.aea._connections.clear()
-        self.aea._connections.append(
-            FakeConnection(envelope, inbox_num, connection_id="fake_connection")
+        self._fake_connection = FakeConnection(
+            envelope, inbox_num, connection_id="fake_connection"
         )
+        # self.aea._connections.clear()
+        self.aea._connections.append(self._fake_connection)
```

Comment:
If you pass the connections to the AEA constructor you can avoid this

Reply:
it's a specific case for tests, and not assume will be used in production